crates/aptos-telemetry/src/service.rs

```diff
@@ -132,46 +150,71 @@ async fn spawn_telemetry_service(
 
     if !force_enable_telemetry() && !telemetry_sender.check_chain_access(chain_id).await {
         warn!(
-            "Aptos telemetry is not sent to the telemetry service because the service is not configured for chain ID {}",
-            chain_id
-        );
+                "Aptos telemetry is not sent to the telemetry service because the service is not configured for chain ID {}",
+                chain_id
+            );
         // Spawn the custom event sender to send to GA4 only.
         // This is a temporary workaround while we deprecate and remove GA4 completely.
         let peer_id = fetch_peer_id(&node_config);
-        tokio::spawn(custom_event_sender(
+        let handle = tokio::spawn(custom_event_sender(
             None,
             peer_id,
             chain_id,
-            node_config,
-            build_info,
+            node_config.clone(),
+            build_info.clone(),
         ));
-        return;
+        info!("Telemetry service for GA4 started!");
+
+        // Check for chain access periodically in case the service is configured later
+        let mut interval = time::interval(Duration::from_secs(CHAIN_ACCESS_CHECK_FREQ_SECS));
+        loop {
+            interval.tick().await;
+            if telemetry_sender.check_chain_access(chain_id).await {
+                handle.abort();
+                break;
+            }
+        }
```

Comment:
nit: an info/warn log when the loop finally breaks? Without this, the logs suggest the chain access failed, but it eventually succeeded.

Reply:
There is a log that says that service started below, but I get your point. Added.